### PR TITLE
Advanced Capabilities week 2 day 3 update

### LINKED ISSUE #8: Day 3: Sharing content from and with your app
| | |
|:--|:--- |
| **Definition** | What does it mean to share content between apps? What kind of content are we talking about? |
| **Differentiation** | How can I enable content from my app to be shared elsewhere? How do I set up my app to receive content? |
| **Adoption**| Who is using this capability? |
| **Exercise**| Enable content to be shared from your app. |
| **Related**| Look for other OS integrations throughout [Week 2: Advanced Capabilities](../advanced-capabilities). |
| |

### PR DESCRIPTION
**Pull request by beth-panx**
---

⚠️ Merge/Release date 2/23

---

Closes #8

### CODE REVIEW

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 1]
```diff
@@ -1,15 +1,117 @@
...
+
+![Screenshot of sharing option in Microsoft Edge](_media/day-03/sharing-edge.jpg)
+
+The normal flow of sharing information through a browser involves a sharing dialogue that is usually native to that platform, and provides options of whom and where to share with, which might vary depending on the type of content. For examples like the one mentioned above, where we are sharing a web page (or a URL), then any application that handles text or links can act as a target. It may be that you wish to share this page with a friend via instant messaging, or email the link to a co-worker; these are just two of many different options that we use on a daily basis. This is how sharing looks on Android:
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:24]:
```suggestion
Sharing information through a web browser usually involves a sharing dialog box that is native to the operating system being used. The dialog provides options of whom and where to share with, which might vary depending on the type of content. In the example mentioned above, where we are sharing a web page (or a URL), then any application that handles text or links can act as a target that can receive the shared content. It may be that you wish to share this page with a friend via instant messaging, or email the link to a co-worker; these are just two of many different options that we use on a daily basis. This is how sharing looks on Android:
```

Comment (captainbrosset) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:24]: These are just suggestions that I feel make reading this part easier. But feel free to drop.

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 2]
```diff
@@ -1,15 +1,117 @@
...
+
+![Screenshot of sharing UX in Microsoft Edge on Android](_media/day-03/sharing-mobile.jpg)
+
+And this is how it looks on macOS.
+
+![Screenshot of the sharing UI in Microsoft Edge on macOS](_media/day-03/sharing-mac.jpg)
+
+As shown in the images above, the action of sharing invokes the native platform’s UX. This creates a seamless integrated experience independent of where you are sharing from, allowing the user to send information between applications. Now, here is where it gets really interesting.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:32]:
```suggestion
As shown in the images above, the action of sharing invokes the operating system's share dialog. This creates a seamless integrated experience independent of where you are sharing from, allowing the user to send information between applications. Now, here is where it gets really interesting.
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 3]
```diff
@@ -1,15 +1,117 @@
...
+Now that I have laid out the concept of sharing between apps, let's bring PWAs into the mix. We can use the `Web Share API` and `Web Share Target API` to invoke this sharing UX to share specific content, allowing your web app to behave like any other installed app.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:33]:
```suggestion
Now that I have laid out the concept of sharing between apps, let's bring PWAs into the mix. We can use the `Web Share API` and `Web Share Target API` to invoke this sharing dialog and share specific content, allowing your web app to behave like any other installed app.
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 4]
```diff
@@ -1,15 +1,117 @@
...
-| **Adoption** | Who is using this capability? |
-| **Exercise** | Enable content to be shared from your app. |
-| **Related** | Look for other OS integrations throughout [Week 2: Advanced Capabilities](../advanced-capabilities). |
+| **Sharing *from* your application** | Learn how to share content directly from your PWA. |
+| **Sharing *to* your application** |Prepare your PWA to receive shared content. |
+| **Related** | Look for other OS integrations throughout [Week 2: Advanced Capabilities](../advanced-capabilities/). |
+
+---
+
+# Definition
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:18]: Should these headings be h1 level? I think only the page title should, and these should be h2s instead.

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 5]
```diff
@@ -1,15 +1,117 @@
...
+
+# Our demo PWA
+To demonstrate how the `Web Share API` works, I will use the PWinter PWA, available [here](https://aka.ms/learn-PWA/30Days-2.3/github.com/diekus/pwinter). This PWA allows a user to create custom themed PWA logos. These logos can then be saved as SVG files to be shared.
+
+![Screenshot of a PWA that prints PWA logos named "PWinter"](_media/day-03/pwinter.jpg)
+
+# Sharing from your application
+Let’s start by looking at how to share from your web application. We want to be able to share data from the PWA to another app that can handle the type of data being shared. As is the case for many things web, “there’s an API for that”. For this specific scenario, the Web Share API comes to the rescue, and we can use the `navigator.share()` method, which takes as a parameter the data we want to share. This data can be composed of a url, text, title, or an array of files if the system supports it. This will return a promise that resolves if the data is shared. As is the case with advanced powerful APIs, you need to be running off a secure connection to be able to use the `Web Share API`.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:41]: `a url, text, title, or an array of files` makes it seem like those are exclusive choices. Maybe rephrase to `a url, text, title, and even files`.

Comment (captainbrosset) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:41]: I would link the `promise` word in the `return a promise` phrase to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises since this is a series for early-in-career, and we can't assume that people will have read other articles before this one where we might have introduced the concept of promises already.

Comment (captainbrosset) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:41]: `running off a secure connection` might not be so straightforward to our readers, I would add `(i.e. HTTPS)` somewhere in there and link it to https://microsoft.github.io/win-student-devs/#/30DaysOfPWA/core-concepts/02?id=_1-https

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 6]
```diff
@@ -1,15 +1,117 @@
...
+
+## Sharing a link
+The PWinter app can share its URL, making it easy to distribute the address to the app with another user. In order to do so, it displays an icon on the lower right corner that when clicked, invokes the `Web Share API`. The following code shares a title, text and url, and brings up the browser/OS share UI when executed.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:44]: `the browser/OS share UI`: earlier we said that the share dialog was native to the platform. Should we just always use the same phrase to talk about this? Like `operating system share dialog`?

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 7]
```diff
@@ -1,15 +1,117 @@
...
+// Shares a link with the Web Share API
+async function sharePWinter() {
+    const shareData = {
+        title: 'PWinter',
+        text: 'Design your own PWA Logo.',
+        url: 'https://diek.us/pwinter',
+    };
+    await navigator.share(shareData)
+    .then(() => console.log('PWinter shared!'))
+    .catch((error) => console.error(error));
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:56]: Since we're using an `async` function, it might be better to not use `.then` and `.catch` at all. Mixing these 2 styles feels a bit confusing to me:

```javascript
try {
  await navigator.share(shareData);
  console.log('PWinter shared!');
} catch (e) {
  console.lerror(e);
}
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 8]
```diff
@@ -1,15 +1,117 @@
...
+            files: filesArray,
+            title: 'My PWA Logo',
+            text: 'Custom PWA logo from The PWinter.'
+        })
+        .then(() => console.log('Share was successful.'))
+        .catch((error) => console.log('Sharing failed', error));
+        }
+        else {
+            console.log(`System doesn't support sharing.`);
+    }
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:79]: `await` only works inside `async` functions, but there isn't a parent function wrapping this piece of code. Can you please add one? People tend to copy/paste without really thinking too much about it, and in this case the code wouldn't work.

Also, same comment about avoiding to mix `await`-style async code and `.then`-style async code if possible.

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 9]
```diff
@@ -1,15 +1,117 @@
...
+
+```
+
+![Screenshot of the PWinter PWA sharing a custom SVG image](_media/day-03/pwinter-share.jpg)
+
+# Sharing to your application
+Last but definitely not least, you can share information to your PWA. The `Web Share Target API` allows us to do this. This is very handy since you can register the application as to be on the receiving end of a share, hence allowing it to integrate with the OS’s share UX. In the previous image, you can spot that the SVG file created by the user can be shared with a set of apps listed in the bottom part of the OS’s share UX; notably, SimpleEdit and The PWinter, two of the apps listed above, are themselves PWAs that can handle image files.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:86]:
```suggestion
Last but definitely not least, you can share information from other applications to your PWA. The `Web Share Target API` allows us to do this. This is very handy since you can register your PWA to be on the receiving end of a share, hence allowing it to integrate with the OS’s share dialog. In the previous image, you can spot that the SVG file created by the user can be shared with a set of apps listed in the bottom part of the OS’s share dialog; notably, SimpleEdit and The PWinter, two of the apps listed above, are themselves PWAs that can handle image files.
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 10]
```diff
@@ -1,15 +1,117 @@
...
+
+The way to register a PWA as a share target is through the manifest file by adding a “share_target” entry to it. In order to accept simple data, such as a link or text, the entry would look like the following:
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:88]:
```suggestion
The way to register a PWA as a share target is through the web app manifest file by adding the `share_target` member to it. In order to accept simple data, such as a link or text, the code would look like the following:
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 11]
```diff
@@ -1,15 +1,117 @@
...
+      "method": "GET",
+      "enctype": "application/x-www-form-urlencoded",
+      "params": {
+        "title": "title",
+        "text": "text",
+        "url": "url"
+      }
+    }
+```
+Something important to note is that the action key defines which part of the PWA will handle the incoming data. For the demo PWA, this addition to the manifest file indicates that a link will be received and processed in the `index.html?shared-color` url. This file has code in place that knows that when this url is called, it will try to parse the incoming link to detect any hex colour code; this will set the letters of the PWA logo to that specific colour.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:102]:
```suggestion
Something important to note is that the `action` property defines which part of the PWA will handle the incoming data. For the demo PWA, this addition to the manifest file indicates that a link will be received and processed by the `index.html?shared-color` url. This file has code in place that knows that when this url is called, it will try to parse the incoming link to detect any hex color code; this will set the letters of the PWA logo to that specific color.
```

(color, colour 😃 I imagine we have more US-based readers than UK-based readers, but up to you).

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 12]
```diff
@@ -1,15 +1,117 @@
...
+
+![Screenshot of the PWA PWinter acting as a share target for a link shared through Microsoft Edge](_media/day-03/pwinter-share-target.jpg)
+
+The incoming information can change the target app, and also accept files. For this, you must set the method key value to `POST`, the enctype field to `multipart/form-data`, and a files array for any files that might be attached. When the user shares to a PWA, the browser will open up the app on the action url and process the data accordingly.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:106]: This reads a little odd. The above code snippet uses GET and a different enctype. So I'm assuming that you want to rephrase to something like this:

`If the shared data received by your PWA is supposed to change the app, then you must set the method property to POST and ...`

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 13]
```diff
@@ -1,15 +1,117 @@
...
+The `Web Share API` is currently supported on Windows, macOS, Linux, Android and iOS. There are some engines that do not have support, so be ready to check and accommodate your app accordingly. `Web Share Target API` relies on the manifest and installation of a PWA which means it is supported on Chromium-based browsers.
+
+Now you are ready to share and continue with tomorrow's post in the [series](https://aka.ms/learn-pwa/30Days-blog) where you'll learn about display modes! 
+
+
+# Resources
+* [Share content with other apps](https://aka.ms/learn-PWA/30Days-2.3/developer.mozilla.org/en-US/docs/Web/API/Navigator/share) at Microsoft Docs.
+* [Web Share API](https://aka.ms/learn-PWA/30Days-2.3/developer.mozilla.org/en-US/docs/Web/API/Web_Share_API) at MDN Web Docs
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:117]: Can we please also link to https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/share ?

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 14]
```diff
@@ -1,15 +1,117 @@
...
+
+The incoming information can change the target app, and also accept files. For this, you must set the method key value to `POST`, the enctype field to `multipart/form-data`, and a files array for any files that might be attached. When the user shares to a PWA, the browser will open up the app on the action url and process the data accordingly.
+
+# Summary
+I have explained the two parts of sharing with PWAs. Sharing from the app is achieved by the `Web Share API`. You can share a variety of information with it. You can share to the app by registering it as a share target, with the `Web Share Target API`. This is done through an entry in the manifest file. These 2 advanced capabilities make your PWA integrate seamlessly with your platform, and help create a smooth unified experience.
+The `Web Share API` is currently supported on Windows, macOS, Linux, Android and iOS. There are some engines that do not have support, so be ready to check and accommodate your app accordingly. `Web Share Target API` relies on the manifest and installation of a PWA which means it is supported on Chromium-based browsers.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:110]:
```suggestion
The Web Share API is currently supported on Windows, macOS, Linux, Android and iOS. There are some engines that do not have support, so be ready to check and accommodate your app accordingly. Web Share Target API relies on the Manifest and installation of a PWA which means it is supported on Chromium-based browsers.
```

**Comment by aarongustafson** [comment]
@diekus Patrick & I have reviewed. Please make any final edits.

File: docs/30DaysOfPWA/advanced-capabilities/03.md (hunk 11 above)
Comment (diekus) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:102]: considering in the URL it is spelled as color I think it makes sense to make it consistent.

File: docs/30DaysOfPWA/advanced-capabilities/03.md (hunk 13 above)
Comment (diekus) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:117]: this is already linked in the article!

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 15]
```diff
@@ -1,15 +1,117 @@
...
+The Web Share API is currently supported on Windows, macOS, Linux, Android and iOS. There are some engines that do not have support, so be ready to check and accommodate your app accordingly. Web Share Target API relies on the Manifest and installation of a PWA which means it is supported on Chromium-based browsers.
+
+Now you are ready to share and continue with tomorrow's post in the [series](https://aka.ms/learn-pwa/30Days-blog) where you'll learn about display modes! 
+
+
+# Resources
+* [Share content with other apps](https://aka.ms/learn-PWA/30Days-2.3/developer.mozilla.org/en-US/docs/Web/API/Navigator/share) at Microsoft Docs.
+* [Web Share API](https://aka.ms/learn-PWA/30Days-2.3/developer.mozilla.org/en-US/docs/Web/API/Web_Share_API) at MDN Web Docs
```
Comment (beth-panx) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:117]:
```suggestion
* [Web Share API](https://aka.ms/learn-PWA/30Days-2.3/developer.mozilla.org/en-US/docs/Web/API/Web_Share_API) at MDN Web Docs
* [Share content with other apps](https://aka.ms/learn-PWA/30Days-2.3/docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/share) at Microsoft Edge docs.
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 16]
```diff
@@ -1,15 +1,117 @@
...
+        else {
+            console.log(`System doesn't support sharing.`);
+    }
+
+```
+
+![Screenshot of the PWinter PWA sharing a custom SVG image](_media/day-03/pwinter-share.jpg)
+
+# Sharing to your application
+Last but definitely not least, you can share information from other applications to your PWA. The `Web Share Target API` allows us to do this. This is very handy since you can register your PWA to be on the receiving end of a share, hence allowing it to integrate with the OS’s share dialog. In the previous image, you can spot that the SVG file created by the user can be shared with a set of apps listed in the bottom part of the OS’s share dialog; notably, SimpleEdit and The PWinter, two of the apps listed above, are themselves PWAs that can handle image files.
```
Comment (zateutsch) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:86]:
```suggestion
Last but definitely not least, you can share information from other applications to your PWA. Using the `Web Share Target API`, you can register your PWA to be on the receiving end of a share, allowing it to integrate with the OS’s share dialog. In the previous image, you can spot that the SVG file created by the user can be shared with a set of apps listed in the bottom part of the OS’s share dialog; notably, SimpleEdit and The PWinter, two of the apps listed above, are both PWAs that can handle image files.
```

Comment (zateutsch) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:86]: OS's or OS'? I'm not sure

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 17]
```diff
@@ -1,15 +1,118 @@
...
+
+The way to register a PWA as a share target is through the web app manifest file by adding the `share_target` member to it. In order to accept simple data, such as a link or text, the code would look like the following:
```
Comment (zateutsch) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:88]:
```suggestion
You can register a PWA as a share target by adding the `share_target` member to the app's web manifest file. In order to accept simple data, such as a link or text, the code would look like the following:
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 18]
```diff
@@ -1,15 +1,118 @@
...
+
+![Screenshot of the PWA PWinter acting as a share target for a link shared through Microsoft Edge](_media/day-03/pwinter-share-target.jpg)
+
+The incoming information can change the target app, and also accept files. For this, you must set the method key value to `POST`, the enctype field to `multipart/form-data`, and a files array for any files that might be attached. When the user shares to a PWA, the browser will open up the app on the action url and process the data accordingly.
+
+# Summary
+I have explained the two parts of sharing with PWAs. Sharing from the app is achieved by the Web Share API. You can share a variety of information with it. You can share to the app by registering it as a share target, with the Web Share Target API. This is done through an entry in the manifest file. These 2 advanced capabilities make your PWA integrate seamlessly with your platform, and help create a smooth, unified experience.
```
Comment (zateutsch) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:109]:
```suggestion
I have explained the two parts of sharing with PWAs. Sharing from the app is achieved by the Web Share API. You can share a variety of information with it. You can share to the app by registering it as a share target, with the Web Share Target API. This is done through an entry in the manifest file. These two advanced capabilities help your PWA integrate seamlessly with your platform, and help create a smooth, unified sharing experience.
```

**Review by zateutsch — APPROVED** [review]
Minor suggestions. Nothing pressing.

Thanks for the article, Diego!

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 19]
```diff
@@ -1,15 +1,123 @@
...
+            text: 'Custom PWA logo from The PWinter.'
+        })
+        .then(() => console.log('Share was successful.'))
+        .catch((error) => console.log('Sharing failed', error));
+        }
+        else {
+            console.log(`System doesn't support sharing.`);
+    }
+};
+```
```
Comment (beth-panx) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:65]:
```suggestion
```javascript
// Shares a link with the Web Share API
async function shareLogo() {
    let fileToShare = createFileForSharing(preparePWALogoforSVG());
    let filesArray = [];
    filesArray[0] = fileToShare;
    if (navigator.canShare && navigator.canShare({ files: filesArray })) {
        try {
            await navigator.share({
                files: filesArray,
                title: 'My PWA Logo',
                text: 'Custom PWA logo from The PWinter.'
            });
            console.log('Share was successful.');
        } catch (error) {
            console.log('Sharing failed', error);
        }
    } else {
        console.log(`System doesn't support sharing.`);
    }
};
```
```

Comment (beth-panx) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:65]: @aarongustafson @diekus How does this look?

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 20]
```diff
@@ -1,15 +1,123 @@
...
+            text: 'Custom PWA logo from The PWinter.'
+        })
+        .then(() => console.log('Share was successful.'))
+        .catch((error) => console.log('Sharing failed', error));
+      }
+        else {
+            console.log(`System doesn't support sharing.`);
+    }
+
+```
```
Comment (beth-panx) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:87]: What's the difference between this and the function starts on line 46? @diekus

Comment (beth-panx) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:87]:
```suggestion
```javascript
// Shares a link with the Web Share API
async function shareLogo() {
    let fileToShare = createFileForSharing(preparePWALogoforSVG());
    let filesArray = [];
    filesArray[0] = fileToShare;
    if (navigator.canShare && navigator.canShare({ files: filesArray })) {
        try {
            await navigator.share({
                files: filesArray,
                title: 'My PWA Logo',
                text: 'Custom PWA logo from The PWinter.'
            });
            console.log('Share was successful.');
        } catch (error) {
            console.log('Sharing failed', error);
        }
    } else {
        console.log(`System doesn't support sharing.`);
    }
};
```
```

Comment (beth-panx) [reply docs/30DaysOfPWA/advanced-capabilities/03.md:87]: I also updated this function.

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 21]
```diff
@@ -1,15 +1,123 @@
...
+
+![Screenshot of sharing UX in Microsoft Edge on Android](_media/day-03/sharing-mobile.jpg)
+
+And this is how it looks on macOS:
+
+![Screenshot of the sharing UI in Microsoft Edge on macOS](_media/day-03/sharing-mac.jpg)
+
+As shown in the images above, the action of sharing invokes the operating system's share dialog. This creates a seamless, integrated experience, independent of where you are sharing from, allowing the user to send information between applications. Now, here is where it gets really interesting.
+Now that I have laid out the concept of sharing between apps, let's bring PWAs into the mix. We can use the `Web Share API` and `Web Share Target API` to invoke this sharing dialog and share specific content, allowing your web app to behave like any other installed app.
```
Comment (beth-panx) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:33]:
```suggestion

Now that I have laid out the concept of sharing between apps, let's bring PWAs into the mix. We can use the `Web Share API` and `Web Share Target API` to invoke this sharing dialog and share specific content, allowing your web app to behave like any other installed app.
```

File: docs/30DaysOfPWA/advanced-capabilities/03.md [hunk 22]
```diff
@@ -1,15 +1,123 @@
...
+      }
+        else {
+            console.log(`System doesn't support sharing.`);
+    }
+
+```
+
+![Screenshot of the PWinter PWA sharing a custom SVG image](_media/day-03/pwinter-share.jpg)
+
+## Sharing to your application
```
Comment (beth-panx) [inline docs/30DaysOfPWA/advanced-capabilities/03.md:91]:
```suggestion
## Sharing _to_ your application
```

**Review by aarongustafson — APPROVED** [review]
